File: app/src/main/java/com/cvapp/assignment/utils/LocalDataStorage.java
package com.cvapp.assignment.utils;

import android.content.Context;
import android.util.Log;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;

import static android.content.Context.MODE_PRIVATE;
import static com.cvapp.assignment.utils.Constants.FILE_NAME;

/**
 * Created by Mathavan_K on 7/8/2019.
 */

public class LocalDataStorage {

    private Context context;

    private static LocalDataStorage single_instance = null;

    JSONObject jsonObject = new JSONObject();

    private LocalDataStorage(Context ctx) {
        this.context = ctx;
    }
    /**
     *  return the singleton instance of LocalDataStorage
     * @return
     */
    public static LocalDataStorage getInstance(Context ctx) {
        if (single_instance == null)
            single_instance = new LocalDataStorage(ctx);

        return single_instance;
    }

    /**
     * save data into cloud storage as json file
     * @param jsonObject
     */
    public void saveData(String jsonObject){
        FileOutputStream fos = null;
        try {
            fos = context.openFileOutput(FILE_NAME, MODE_PRIVATE);
            fos.write(jsonObject.toString().getBytes());
            Log.v("Filename----->", "Filename path---->" + context.getFilesDir() + "/" + FILE_NAME);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            if (fos != null) {
                try {
                    fos.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
    }

    /**
     * get the filepath
     * @return
     */
    public String getFilePath(){
        return context.getFilesDir()+"/"+FILE_NAME;
    }

    /**
     * delete the file
     */
    public void deleteFileFromLocal(){
           context.deleteFile(FILE_NAME);
    }

}

File: app/src/main/java/com/cvapp/assignment/utils/LocalDataStorage.kt
package com.cvapp.assignment.utils

import android.content.Context
import android.util.Log

import org.json.JSONException
import org.json.JSONObject

import java.io.FileNotFoundException
import java.io.FileOutputStream
import java.io.IOException

import android.content.Context.MODE_PRIVATE
import com.cvapp.assignment.utils.Constants.Companion.FILE_NAME

/**
 * Created by Mathavan_K on 7/8/2019.
 */

class LocalDataStorage private constructor(private val context: Context) {

    /**
     * get the filepath
     * @return
     */
    val filePath: String
        get() = context.filesDir.toString() + "/" + FILE_NAME

    /**
     * save data into cloud storage as json file
     * @param jsonObject
     */
    fun saveData(jsonObject: String) {
        var fos: FileOutputStream? = null
        try {
            fos = context.openFileOutput(FILE_NAME, MODE_PRIVATE)
            fos!!.write(jsonObject.toByteArray())
            Log.v("Filename----->", "Filename path---->" + context.filesDir + "/" + FILE_NAME)
        } catch (e: FileNotFoundException) {
            e.printStackTrace()
        } catch (e: IOException) {
            e.printStackTrace()
        } finally {
            if (fos != null) {
                try {
                    fos.close()
                } catch (e: IOException) {
                    e.printStackTrace()
                }

            }
        }
    }

    /**
     * delete the file
     */
    fun deleteFileFromLocal() {
        context.deleteFile(FILE_NAME)
    }

    companion object {

        private var single_instance: LocalDataStorage? = null
        /**
         * return the singleton instance of LocalDataStorage
         * @return
         */
        fun getInstance(ctx: Context): LocalDataStorage {
            if (single_instance == null)
                single_instance = LocalDataStorage(ctx)

            return single_instance as LocalDataStorage
        }
    }

}
